liblangutil/SourceLocation.h

```diff
@@ -119,4 +119,23 @@ SourceLocation parseSourceLocation(
 /// Stream output for Location (used e.g. in boost exceptions).
 std::ostream& operator<<(std::ostream& _out, SourceLocation const& _location);
 
+
+/**
+ * Alternative, line-column-based representation for source locations.
+ * Both line and column are zero-based.
+ * If used as a range, the second location is considered exclusive.
+ * Negative values are invalid.
+ */
+struct LineColumn
+{
+	/// Line value, can be between zero and number of `\n` characters in the source file.
+	int line = {-1};
+	/// Column value, can be between zero and number of characters in the line (inclusive).
+	int column = {-1};
```

Comment:
It's just an `int`. Why use an initializer list?
```suggestion
	/// Line value, can be between zero and number of `\n` characters in the source file.
	int line = -1;
	/// Column value, can be between zero and number of characters in the line (inclusive).
	int column = -1;
```